### ISSUE TITLE
Set lhttpc_pool field in aws_config generates no_proc error

### ISSUE THREAD

**Issue by posilva** [erlcloud/erlcloud#380]
Hi,

when setting the `lhttpc_pool` field (an atom()) in the `#aws_config{}` to a value like `ddb_pool` it causes an exception: 
```erlang
** exception exit: {noproc,{gen_server,call,
                                       [ddb_pool,
                                        {socket,<0.160.0>,"ec2.amazonaws.com",443,true},
                                        infinity]}}

```
the reason is in [erlcloud_httpc.erl#L49](https://github.com/erlcloud/erlcloud/blob/master/src/erlcloud_httpc.erl#L49) we are adding an extra option to the request with the pool name, but in [lhttpc_manager.erl#L211](https://github.com/erlcloud/lhttpc/blob/master/src/lhttpc_manager.erl#L211) is expected to be the option `pool_ensure` to force to create the lhttpc pool.

Two fix this I see 2 (simple) options:

1- When adding the option `{pool, Pool}` we also add `{pool_ensure, true}` in this case we do not break the interface, and it only creates the pool if no pool process exists.

2- Extend the `lhttpc_pool` field to accommodate a extra options or create a new `lhttpc_pool_opts` field to allow to set the ensure_pool option, or other that may be necessary in certain cases.

The work around I have now is set http_client field to a anonymous function that sets the pools parameters.

Please tell me what is the preferred approach, i prefer the option 1., and i can do PR.

Here are the steps to reproduce: 

```erlang
application:ensure_all_started(erlcloud).
rr("include/erlcloud_aws.hrl").
{ok, Conf} = erlcloud_aws:profile().
NewConf=Conf#aws_config{lhttpc_pool=test_pool}.
erlcloud_ec2:describe_instances(NewConf).

```

Cheers

**Comment by nalundgaard** [comment]
I'd personally prefer that we document the need for you to own your own pool, if not using the default one. It seems to me that if you're using the feature, you probably want much more control of your request pool. I think it'd be better to have people manage their pooling rather than more deeply integrate erlcloud into lhttpc as an HTTP client.

**Comment by posilva** [comment]
Hi @nalundgaard,

so whats the point of having the option there that just do not work. If we opt for my suggestion 1. then we can document the degree of control that is offered: "If someone opts for this config then will be created a new pool with the defined or default size of the pools in `lhttpc`." and we can also always  document what is the alternative if we want better control (as you suggested).

In my case I am also adjusting the `lhttpc` default pool size using `lhttpc` app.config settings. 

I can contribute for this if you think that it's worth it.

**Comment by nalundgaard** [comment]
@posilva I would say that the option does work, it just requires that you reference a pool you have created separately. I don't consider that broken.

That said, I see where you're coming from, and I am not opposed to your option 1—adding `{ensure_pool, true}` if `lhttpc_pool /= undefined`, and documenting the behavior. That seems like a good compromise: if you want to tune your pools yourself, create them yourself, but if you just want some pools with default settings, you can just name them and they will be made.

If you want to contribute that change, I'm ok with it. @motobob / @waisbrot / @smilerlee any opposition to that?

**Comment by posilva** [comment]
Is the aws_config record definition the right place to document

**Comment by nalundgaard** [comment]
Resolved by #382
